react-app/src/components/Layout/index.tsx

```diff
@@ -254,7 +257,7 @@ export const Layout = () => {
         <SimplePageContainer>
           <Banner />
         </SimplePageContainer>
-        <Outlet />
+        {error ? <ErrorPage /> : <Outlet />}
```

Comment:
you might want to refactor this to use the React Error Boundary convention 

https://react.dev/reference/react/useTransition#displaying-an-error-to-users-with-error-boundary